File: src/main/java/ro/nicolaemariusghergu/queryit/service/TruckService.java
package ro.nicolaemariusghergu.queryit.service;

import org.springframework.lang.NonNull;
import ro.nicolaemariusghergu.queryit.model.Truck;

import java.util.List;
import java.util.Optional;

public interface TruckService {

    @NonNull
    Optional<Truck> findById(@NonNull Long id);

    @NonNull
    List<Truck> findAll();

    Optional<Truck> findBySerialNumber(String serialNumber);

    <S extends Truck> S save(S entity);

    <S extends Truck> List<S> saveAll(Iterable<S> entities);
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/service/TruckService.kt
package ro.nicolaemariusghergu.queryit.service

import org.springframework.http.ResponseEntity
import ro.nicolaemariusghergu.queryit.dto.TruckDto

interface TruckService {
    open fun findTruckById(id: Long?): ResponseEntity<TruckDto?>?
    open fun getTrucks(): ResponseEntity<MutableList<TruckDto?>?>?
    open fun addTruck(truckDto: TruckDto?): ResponseEntity<Long?>?
    open fun getTruckBySerialNumber(serialNumber: String?): ResponseEntity<TruckDto?>?
    open fun deleteTruckById(id: Long?): ResponseEntity<Long?>?
    open fun updateTruck(truckDto: TruckDto?): ResponseEntity<TruckDto?>?
}